data/src/main/java/io/gnosis/data/repositories/TransactionRepository.kt

```diff
@@ -137,48 +143,76 @@ class TransactionRepository(
         val to = transaction.dataDecoded?.parameters?.getValueByName("to")?.asEthereumAddress() ?: Solidity.Address(BigInteger.ZERO)
         val value = BigInteger.ONE
 
+        val tokenInfo = serviceTokenInfo(transaction)
+
         return Transaction.Transfer(
             transaction.status(safeInfo),
+            transaction.confirmations?.size ?: 0,
             to,
             from,
             value,
-            transaction.executionDate?.formatBackendDate(),
-            FAKE_ERC721_TOKEN_INFO // TODO: find out correct token data source
+            transaction.bestAvailableDate(),
+            tokenInfo,
+            transaction.nonce
         )
     }
 
+    private fun serviceTokenInfo(transfer: TransferDto): ServiceTokenInfo =
+        transfer.tokenInfo
+            ?: transfer.tokenAddress?.let { address ->
+                Log.w("NFT", "transfer.tokenAddress: ${transfer.tokenAddress}")
```

Comment:
Log -> Timber

Reply:
Timber is not (yet) available in data. But also the log statement was only for debugging. So I deleted it.